### ISSUE TITLE
nagios segfaults on invalid free()

### ISSUE THREAD

**Issue by nateucar** [NagiosEnterprises/nagioscore#469]
-  Use a clear and concise title for the Issue to identify the problem accurately
Nagios crashes during startup
- Describe the bug with as much detail as you can
Backtrace is below.
- Include the version of the project containing the bug you're reporting
Git revision on master: a70358b97c4d597bec609517e70b5607e362b5be 
compiled from source on ppc64le in default debian docker image
- Include your operating system information (uname -a)
Linux nagios 4.9.0-5-powerpc64le #1 SMP Debian 4.9.65-3+deb9u2 (2018-01-04) ppc64le GNU/Linux
- Include a list of third party modules that are installed and/or loaded
None
- Explain the behavior you expected to see (and why) vs. what actually happened
It crashes on startup
- Describe the exact steps that reproduce the problem
Call `/usr/local/nagios/bin/nagios /etc/nagios/nagios.cfg`
- Explain when the problem started happening: was it after an upgrade? or was it always present?
This just started with latest nagios master. We didn't track the last git rev we were using.
- Define how reliably you can reproduce the bug
Every time it starts

Here is the backtrace
> Successfully launched command file worker with pid 5291
> *** Error in `/usr/local/nagios/bin/nagios': free(): invalid pointer: 0x000000010ce951e0 ***
> ======= Backtrace: =========
> /lib/powerpc64le-linux-gnu/libc.so.6(+0x82a68)[0x3fff9e6a2a68]
> /lib/powerpc64le-linux-gnu/libc.so.6(+0x8c118)[0x3fff9e6ac118]
> /lib/powerpc64le-linux-gnu/libc.so.6(+0x8ccec)[0x3fff9e6accec]
> /usr/local/nagios/bin/nagios(clear_host_macros_r+0x80)[0x10ce09230]
> /usr/local/nagios/bin/nagios(clear_volatile_macros_r+0x58)[0x10ce09988]
> /usr/local/nagios/bin/nagios(run_async_host_check+0x400)[0x10cde1e90]
> /usr/local/nagios/bin/nagios(run_scheduled_host_check+0xc0)[0x10cde2270]
> /usr/local/nagios/bin/nagios(handle_timed_event+0x6ec)[0x10cdfcd6c]
> /usr/local/nagios/bin/nagios(event_execution_loop+0x448)[0x10cdfd8c8]
> /usr/local/nagios/bin/nagios(main+0x1048)[0x10cdd51c8]
> /lib/powerpc64le-linux-gnu/libc.so.6(+0x2289c)[0x3fff9e64289c]
> /lib/powerpc64le-linux-gnu/libc.so.6(__libc_start_main+0xb8)[0x3fff9e642ab8]
> ======= Memory map: ========
> 10cdc0000-10ce80000 r-xp 00000000 09:7f 3827736610                       /usr/local/nagios/bin/nagios
> 10ce80000-10ce90000 r--p 000b0000 09:7f 3827736610                       /usr/local/nagios/bin/nagios
> 10ce90000-10cea0000 rw-p 000c0000 09:7f 3827736610                       /usr/local/nagios/bin/nagios
> 1003ad60000-1003ae40000 rw-p 00000000 00:00 0                            [heap]
> 3fff8d940000-3fff9cd00000 rw-p 00000000 00:00 0 
> 3fff9cd00000-3fff9cd10000 r-xp 00000000 09:7f 1722736659                 /lib/powerpc64le-linux-gnu/libnss_files-2.24.so
> 3fff9cd10000-3fff9cd20000 r--p 00000000 09:7f 1722736659                 /lib/powerpc64le-linux-gnu/libnss_files-2.24.so
> 3fff9cd20000-3fff9cd30000 rw-p 00010000 09:7f 1722736659                 /lib/powerpc64le-linux-gnu/libnss_files-2.24.so
> 3fff9cd30000-3fff9cd40000 r-xp 00000000 09:7f 1722736663                 /lib/powerpc64le-linux-gnu/libnss_nis-2.24.so
> 3fff9cd40000-3fff9cd50000 r--p 00000000 09:7f 1722736663                 /lib/powerpc64le-linux-gnu/libnss_nis-2.24.so
> 3fff9cd50000-3fff9cd60000 rw-p 00010000 09:7f 1722736663                 /lib/powerpc64le-linux-gnu/libnss_nis-2.24.so
> 3fff9cd60000-3fff9cd80000 r-xp 00000000 09:7f 1722736653                 /lib/powerpc64le-linux-gnu/libnsl-2.24.so
> 3fff9cd80000-3fff9cd90000 r--p 00010000 09:7f 1722736653                 /lib/powerpc64le-linux-gnu/libnsl-2.24.so
> 3fff9cd90000-3fff9cda0000 rw-p 00020000 09:7f 1722736653                 /lib/powerpc64le-linux-gnu/libnsl-2.24.so
> 3fff9cda0000-3fff9cdb0000 r-xp 00000000 09:7f 1722736655                 /lib/powerpc64le-linux-gnu/libnss_compat-2.24.so
> 3fff9cdb0000-3fff9cdc0000 r--p 00000000 09:7f 1722736655                 /lib/powerpc64le-linux-gnu/libnss_compat-2.24.so
> 3fff9cdc0000-3fff9cdd0000 rw-p 00010000 09:7f 1722736655                 /lib/powerpc64le-linux-gnu/libnss_compat-2.24.so
> 3fff9cdd0000-3fff9e5f0000 rw-p 00000000 00:00 0 
> 3fff9e5f0000-3fff9e600000 r-xp 00000000 09:7f 1714751592                 /lib/powerpc64le-linux-gnu/libdl-2.24.so
> 3fff9e600000-3fff9e610000 r--p 00000000 09:7f 1714751592                 /lib/powerpc64le-linux-gnu/libdl-2.24.so
> 3fff9e610000-3fff9e620000 rw-p 00010000 09:7f 1714751592                 /lib/powerpc64le-linux-gnu/libdl-2.24.so
> 3fff9e620000-3fff9e7c0000 r-xp 00000000 09:7f 1714751582                 /lib/powerpc64le-linux-gnu/libc-2.24.so
> 3fff9e7c0000-3fff9e7d0000 r--p 00190000 09:7f 1714751582                 /lib/powerpc64le-linux-gnu/libc-2.24.so
> 3fff9e7d0000-3fff9e7e0000 rw-p 001a0000 09:7f 1714751582                 /lib/powerpc64le-linux-gnu/libc-2.24.so
> 3fff9e7e0000-3fff9e7f0000 r-xp 00000000 09:7f 2206368770                 /usr/lib/powerpc64le-linux-gnu/libltdl.so.7.3.1
> 3fff9e7f0000-3fff9e800000 r--p 00000000 09:7f 2206368770                 /usr/lib/powerpc64le-linux-gnu/libltdl.so.7.3.1
> 3fff9e800000-3fff9e810000 rw-p 00010000 09:7f 2206368770                 /usr/lib/powerpc64le-linux-gnu/libltdl.so.7.3.1
> 3fff9e810000-3fff9e8d0000 r-xp 00000000 09:7f 1722736644                 /lib/powerpc64le-linux-gnu/libm-2.24.so
> 3fff9e8d0000-3fff9e8e0000 r--p 000b0000 09:7f 1722736644                 /lib/powerpc64le-linux-gnu/libm-2.24.so
> 3fff9e8e0000-3fff9e8f0000 rw-p 000c0000 09:7f 1722736644                 /lib/powerpc64le-linux-gnu/libm-2.24.so
> 3fff9e8f0000-3fff9e900000 rw-p 00000000 00:00 0 
> 3fff9e900000-3fff9e920000 r-xp 00000000 00:00 0                          [vdso]
> 3fff9e920000-3fff9e950000 r-xp 00000000 09:7f 1714751545                 /lib/powerpc64le-linux-gnu/ld-2.24.so
> 3fff9e950000-3fff9e960000 r--p 00020000 09:7f 1714751545                 /lib/powerpc64le-linux-gnu/ld-2.24.so
> 3fff9e960000-3fff9e970000 rw-p 00030000 09:7f 1714751545                 /lib/powerpc64le-linux-gnu/ld-2.24.so
> 3ffffc8b0000-3ffffc8e0000 rw-p 00000000 00:00 0                          [stack]
> Aborted

**Comment by nateucar** [comment]
Reverting to tag nagios-4.3.4 fixed the crashing at startup. No changes in config or anything else.

**Comment by hedenface** [comment]
Thanks for the detailed report. I'll take a look and let you know my
findings.

**Comment by hedenface** [comment]
I actually bet this was due to a commit that I issued to attempt and try and fix something that I reckon didn't need fixing. 

If you issue a:

```
git revert b5af4c
```

on the current 4.4.0 (master) branch - does the problem go away?

**Comment by nateucar** [comment]
The revert works. Nagios loads fine from the master branch with that reverted.

**Comment by hedenface** [comment]
I'll make sure this gets reverted for the 4.4 release.

**Comment by hedenface** [comment]
Nope. That was my pull/push merge. Fixed here: https://github.com/NagiosEnterprises/nagioscore/commit/650408faa4397270ef571bc0a5e8741a2eb21967
